examples/mugen/retrieval/video_clip.py

```diff
@@ -76,36 +87,153 @@ def forward(self, input_ids: torch.Tensor) -> torch.Tensor:
 
 
 class VideoEncoder(nn.Module, PretrainedMixin):
-    """Encode videos to a fixed size vector.
+    """Encode videos to the last layer before the fully-connected layer of S3D.
 
     Adapted from MUGEN's video encoder
         (https://github.com/mugen-org/MUGEN_baseline/blob/main/lib/models/videoclip/modules.py)
 
     Args:
         pretrained (bool): whether to use a pretrained model or not.
             Defaults to True.
+        trainable (bool): whether the encoder's weights should be trainable (not frozen).
+            Defaults to True.
+            Warning: If ``pretrained`` is False, ``trainable`` will act as True regardless of the supplied value.
         pretrain_path (str): local path or remote URL to pretrained weights.
             Defaults to ``PRETRAINED_S3D_KINETICS400_URL``, the weights MUGEN used from
             pretraining S3D on Kinetics 400. Ignored if ``pretrained`` is False.
 
     Inputs:
         x (Tensor): batch of videos with dimensions (batch, channel, time, height, width)
+            Size of ``channel`` dimension must be 3.
+
     """
 
     def __init__(
         self,
         pretrained: bool = True,
```

Comment:
if pretrained is True, the user likely don't want to train the model. It looks over complicated here. Apart from MUGEN, have you seen elsewhere that requires both flags?

Reply:
as far as i know it's common to use a pretrained model and then continue training it, but I'm open to other suggestions for simplifying the API if possible